packages/app/src/app/overmind/actions.ts

```diff
@@ -79,13 +79,15 @@ type ModalName =
   | 'signInForTemplates'
   | 'userSurvey';
 
-export const modalOpened: Action<{ modal: ModalName; message?: string }> = (
-  { state, effects },
-  { modal, message }
-) => {
+export const modalOpened: Action<{
+  modal: ModalName;
+  message?: string;
+  itemId?: string;
+}> = ({ state, effects }, { modal, message, itemId }) => {
   effects.analytics.track('Open Modal', { modal });
   state.currentModalMessage = message;
   state.currentModal = modal;
+  state.preferences.itemId = itemId;
```

Comment:
This means `currentModal` needs to be set to `'preferences'`, so not sure if this is the right approach 🤔

Reply:
Yeah, it's not my favorite. 

Added a check for `currentModal` similar to `modalClosed`. Ideally I'd like to send `data` which is deepmerged in to the object

```js
modalOpened({
  modal: 'preferences',
  data: { itemId: 'paymentInfo' }
});
```

But this would only work for modals, not the rest of the overmind actions. Would talk to @christianalfoni for best practices in a different pull request

Reply:
I think adding a `data` key that will be merged is indeed a much cleaner way to accomplish this 🙂